### ISSUE TITLE
GwUsersWebManager pattern tries to re-register the database models even if the GwUsersWebManager plugin activation does it.

### ISSUE THREAD

**Issue by amhaske** [useblocks/groundwork-users#1]
Hi  @danwos ,
I have been using the GwUsersPattern for registering the users in the project, for which the plugin i use inherits from GwUsersPattern.
I have also set my project to run and activate the GwUsersWebManager plugin so that i can get web views and menu entries for the registered users.
But the problem i am facing here is when i want to register a user using `self.users.register()`  i need my plugin to inherit from the GwUsersPattern.
When i do this i get an exception saying `Database users_db already registered  by GwUsersWebManager`

**Please find the complete traceback below:**

`Traceback (most recent call last):
  File "D:\UseBlocks\GwEnv\Scripts\milestone_manager-script.py", line 11, in <module>
    load_entry_point('milestone-manager', 'console_scripts', 'milestone_manager')()
  File "d:\useblocks\groundworkprojects\ascgit307.milestone_manager\milestone_manager\applications\miles
tone_manager_app.py", line 18, in start_app
    MILESTONE_MANAGER_APP().start()
  File "d:\useblocks\groundworkprojects\ascgit307.milestone_manager\milestone_manager\applications\miles
tone_manager_app.py", line 13, in start
    self.app.plugins.activate(self.app.config.get("PLUGINS", None))
  File "d:\useblocks\gwenv\lib\site-packages\groundwork\pluginmanager.py", line 156, in activate
    self.initialise_by_names([plugin_name])
  File "d:\useblocks\gwenv\lib\site-packages\groundwork\pluginmanager.py", line 76, in initialise_by_nam
es
    self.initialise(plugin_class.clazz, plugin_name)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork\pluginmanager.py", line 87, in initialise
    plugin_instance = clazz(app=self._app, name=name)
  File "d:\useblocks\groundworkprojects\ascgit307.milestone_manager\milestone_manager\plugins\milestone_
manager_db_plugin\milestone_manager_db_plugin.py", line 13, in __init__
    super().__init__(app, **kwargs)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork\patterns\gw_commands_pattern.py", line 50, in __
init__
    super(GwCommandsPattern, self).__init__(*args, **kwargs)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_web\patterns\gw_web_db_admin_pattern\gw_web_db_a
dmin_pattern.py", line 18, in __init__
    super(GwWebDbAdminPattern, self).__init__(*args, **kwargs)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_web\patterns\gw_web_db_rest_pattern\gw_web_db_re
st_pattern.py", line 17, in __init__
    super(GwWebDbRestPattern, self).__init__(*args, **kwargs)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_web\patterns\gw_web_pattern\gw_web_pattern.py",
line 18, in __init__
    super(GwWebPattern, self).__init__(*args, **kwargs)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_users\patterns\gw_users_pattern\gw_users_pattern
.py", line 35, in __init__
    self.users_db = self.databases.register(db_name, db_url, db_description)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_database\patterns\gw_sql_pattern.py", line 50, i
n register
    return self.app.databases.register(database, database_url, description, self.plugin)
  File "d:\useblocks\gwenv\lib\site-packages\groundwork_database\patterns\gw_sql_pattern.py", line 91, i
n register
    database, self._databases[database].plugin.name))
groundwork_database.patterns.gw_sql_pattern.DatabaseExistException: Database users_db already registered
 by GwUsersWebManager
`

**Comment by danwos** [comment]
Okay, the database registration happens everytime __init__ gets called.

```
self.users_db = self.databases.register(db_name, db_url, db_description)
        models = get_model_classes(self.users_db, self.app)  # User, Permission, Role, Apikey, Domain, Group
        for model in models:
            self.users_db.classes.register(model)
self.users_db.create_all()```

**Comment by amhaske** [comment]
Yes a work-around is possible by using a try except statement while registering the database, so that if the database is already registered by GwUsersWebManager plugin or by any other plugin then the re-registration of the database can be avoided.

I have made the following changes to the code:

```
# Adding a try except statement in case the models are already registered by some other plugin
        try:
            models = get_model_classes(self.users_db, self.app)  # User, Permission, Role, Apikey, Domain, Group

            for model in models:
                self.users_db.classes.register(model)
            self.users_db.create_all()

        except Exception:
            pass
```
Also a similar change in the signal configuration part is done to avoid re-registration exception:

```
 # Adding a try except statement in case the signals are already registered by some other plugin

        try:
            self.signals.connect(receiver="web_users_activation",
                                 signal="gw_web_loaded",
                                 function=self.configure_web_security,
                                 description="Cares about the correct configuration of "
                                             "flask security for GwUsers",
                                 sender=None)

            self.anonymous_name = "anonymous"
            self.anonymous_role = None
            self.anonymous_user = None

        except Exception:
            pass
```

**Comment by danwos** [comment]
solved by merged pull-request #2.
